### PR TITLE
Combine comparison applications into one

### PR DESCRIPTION
**Pull request by akva2**
Folds compareSummary into compareECL. There is no reason for having two applications (only dev history reasons).

This includes https://github.com/OPM/opm-common/pull/485. merge separate or as one as you see fit..

@alfbr as requested sorry it took me so long (busy ant..).

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=1568 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=1568 please

File: CMakeLists.txt [hunk 1]
```diff
@@ -132,5 +130,34 @@ endmacro (install_hook)
...
+                                   LIBRARIES ${_libs})
+  opm_add_test(test_EclFilesComparator CONDITION ENABLE_ECL_INPUT
+                                       LIBRARIES ${_libs})
+  if(HAVE_DYNAMIC_BOOST_TEST AND NOT BUILD_SHARED_LIBS)
+    set_target_properties(test_compareSummary PROPERTIES
+                          COMPILE_DEFINITIONS BOOST_TEST_DYN_LINK)
+    set_target_properties(test_EclFilesComparator PROPERTIES
+                          COMPILE_DEFINITIONS BOOST_TEST_DYN_LINK)
+  endif()
+  install(TARGETS compareECL DESTINATION bin)
```
[the remark this answers is not in the document]
Comment (akva2) [reply CMakeLists.txt]: library is static, and not meant for installation. afaik this should install the compareECL binary?

File: examples/test_util/compareECL.cpp [hunk 2]
```diff
@@ -136,32 +159,48 @@ int main(int argc, char** argv) {
...
                 checkNumKeywords = true;
                 break;
-            case 'n':
-                throwOnError = false;
-                break;
             case 'k':
                 specificKeyword = true;
                 keyword = optarg;
                 break;
+            case 'K':
```
[the remark this answers is not in the document]
Comment (akva2) [reply examples/test_util/compareECL.cpp]: fixed, thx for spotting.

**Comment by akva2** [comment]
jenkins build this opm-simulators=1568 please

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (blattms) [reply CMakeLists.txt]: Indeed, I missed the `STATIC` and that it is only one binary. Everything is fine.

**Comment by akva2** [comment]
approved, merging.

**Pull request merged**